raiden_contracts/tests/test_channel_unlock.py

```diff
@@ -131,6 +131,111 @@ def test_merkle_root_odd_even_components(
     assert unlocked_amount == 9
 
 
+def test_merkle_tree_components_order(
+        web3,
+        get_accounts,
+        token_network_test_utils,
+        secret_registry_contract,
+        reveal_secrets,
+        token_network,
+        create_settled_channel,
+):
+    network_utils = token_network_test_utils
+    (A, B) = get_accounts(2)
+    types = ['uint256', 'uint256', 'bytes32']
+
+    pending_transfers_tree = get_pending_transfers_tree(web3, [1, 3, 5], [2, 8, 3])
+    reveal_secrets(A, pending_transfers_tree.unlockable)
+
+    channel_identifier = create_settled_channel(
+        A,
+        pending_transfers_tree.locked_amount,
+        pending_transfers_tree.merkle_root,
+        B,
+        0,
+        EMPTY_LOCKSROOT,
+    )
+
+    # Merkle tree lockhashes are ordered lexicographicaly.
+    # If we change the order, we change the computed merkle root.
+    # However, the getMerkleRootAndUnlockedAmount orders neighbouring lockhashes
+    # lexicographicaly, so simple item[i], item[i + 1] swap
+    # will still result in the same merkle root.
+    wrong_order = pending_transfers_tree.transfers
+    wrong_order[1], wrong_order[0] = wrong_order[0], wrong_order[1]
```

Comment:
Should we test this for all pairs? IOW, is this an implementation detail or something that people can/will rely on?

Reply:
It is an implementation detail. We order all the locks lexicographically, so we have a convention of how to order them in the merkle tree to get the proper merkle root. I assume this is better than relying on the time when the transfer was initiated / received etc., which would probably end up in race conditions.

I can test all the pairs if you suggest to do so. I thought the current tests were enough, but it's not a big example anyway, so it is doable.

Reply:
Makes sense, and I'm ok with the tests that are there.